### PR TITLE
Simpler prefix replacement for dest-prefix case

### PR DESCRIPTION
**Pull request by mcg1969**
_Builds on #48._

When `dest-prefix` is supplied, there's really no need to do the more complicated `/usr/bin/env` shebang replacement. I've rearranged the logic in the `self.has_dest` case so that it never calls `rewrite_shebang`. 

Furthermore, `rewrite_shebang` was returning `True` (indicating a replacement was performed) even when the prefix was not found in the shebang itself. This was preventing certain files from being properly fixed, like `python3-config`, which have a generic shebang but contain the string in other locations as well.

The primary reason I want to do this, frankly, is so that we can keep conda's own activate/deactivate scripts around when conda is in the environment. But I think these changes are more worth considering on their own merits, so that proposed change is not in this PR.

### CODE REVIEW

File: conda_pack/core.py [hunk 1]
```diff
@@ -813,8 +813,7 @@ def rewrite_shebang(data, target, prefix):
             new_shebang = '#!/usr/bin/env %s%s' % (executable_name,
                                                    options.decode('utf-8'))
             data = data.replace(shebang, new_shebang.encode('utf-8'))
-
-        return data, True
+            return data, True
```
Comment (mcg1969) [inline conda_pack/core.py]: By moving this return inside the if block, we only return `True` if the shebang is _actually_ changed. There are several scripts that use simple `/bin/sh` shebangs but contain the prefix; those still need to be edited, but the current code fails to do so.

**Comment by dsludwig** [comment]
Delta for this PR: https://github.com/conda/conda-pack/pull/54/files/5d2ea007d77a29c0b951c2a51c2390dcbc8fd9af..df52cb726734650f0be6590105c8891817113ec3

**Comment by mcg1969** [comment]
@jcrist this PR has also been validated in its latest commit.

**Comment by jcrist** [comment]
Seems reasonable. To ease the total review, would you mind merging this in with #48, as they're related.

**Comment by mcg1969** [comment]
Will do.

**Pull request closed without merging**